server/src/api/binary_cache.rs

```diff
@@ -223,7 +223,14 @@ async fn get_nar(
                 });
                 let body = Body::from_stream(stream);
 
-                Ok(body.into_response())
+                let response = Response::builder()
+                    .status(StatusCode::OK)
+                    .header("Content-Type", mime::NAR)
+                    .body(body)
+                    .unwrap()
+                    .into_response();
+
+                Ok(response)
```

Comment:
I think the better way to do this is to return a tuple -- IIRC axum lets you set headers by using a tuple of `(StatusCode, Body, [(HeaderName, HeaderValue)])`

Lemme find an example.

Reply:
Oops, I was wrong, it's `(StatusCode, [(HeaderName, HeaderValue)], Body)`

Reply:
https://github.com/tokio-rs/axum/blob/7c871e9591bd997a04104fb0cfcd52c40ce15ad7/axum-extra/src/response/mod.rs#L68-L74

Here's an example (too bad axum-extra doesn't expose this `mime_response!` macro). Looks like it can even be as simple as `([(HeaderName, HeaderValue), Body)`.

Reply:
Wow I didn't know this was a valid pattern, thanks! Pushed the cleaner way.